flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java

```diff
@@ -49,18 +51,25 @@ public static boolean isStatementComplete(String statement) {
         for (String line : content.split("\n")) {
             if (isEndOfStatement(line)) {
                 buffer.add(line);
-                statements.add(String.join("\n", buffer));
+                statements.add(normalize(buffer));
                 buffer.clear();
             } else {
                 buffer.add(line);
             }
         }
         if (!buffer.isEmpty()) {
-            statements.add(String.join("\n", buffer));
+            statements.add(normalize(buffer));
         }
         return statements;
     }
 
+    private static String normalize(List<String> buffer) {
+        // remove comment lines
+        return buffer.stream()
+                .map(statementLine -> statementLine.replaceAll(BEGINNING_MASK, ""))
```

Comment:
Add `.filter(line -> !line.isEmpty())`?

Reply:
I think we don't need to remove empty lines. Keeping the empty line can help us to keep original line number in the near future.